workers/unity/Assets/Playground/Config/EntityTemplates.cs

```diff
@@ -0,0 +1,115 @@
+        {
+            var clientAttribute = EntityTemplate.GetWorkerAccessAttribute(clientWorkerId);
+
+            var template = new EntityTemplate();
+
+            template.AddComponent(new Position.Snapshot(), clientAttribute);
+            template.AddComponent(new Metadata.Snapshot("Character"), WorkerUtils.UnityGameLogic);
+            template.AddComponent(new PlayerInput.Snapshot(), clientAttribute);
+            template.AddComponent(new Launcher.Snapshot(100, 0), WorkerUtils.UnityGameLogic);
+            template.AddComponent(new Score.Snapshot(), WorkerUtils.UnityGameLogic);
+            template.AddComponent(new CubeSpawner.Snapshot(new List<EntityId>()), WorkerUtils.UnityGameLogic);
+
+            var clientSelfInterest = InterestQuery.Query(Constraint.EntityId(entityId)).FilterResults(new[]
+            {
+                Position.ComponentId, Metadata.ComponentId, TransformInternal.ComponentId, CubeSpawner.ComponentId,
+                Score.ComponentId, Launcher.ComponentId
+            });
+
+            var clientRangeInterest = InterestQuery.Query(Constraint.RelativeCylinder(radius: 25))
+                .FilterResults(new[]
+                {
+                    Position.ComponentId, Metadata.ComponentId, TransformInternal.ComponentId, Collisions.ComponentId,
+                    SpinnerColor.ComponentId, SpinnerRotation.ComponentId, CubeColor.ComponentId, Score.ComponentId,
+                    Launchable.ComponentId
+                });
+
+            var serverSelfInterest = InterestQuery.Query(Constraint.EntityId(entityId)).FilterResults(new[]
+            {
+                Position.ComponentId, Metadata.ComponentId, TransformInternal.ComponentId, Score.ComponentId
+            });
+
+            var serverRangeInterest = InterestQuery.Query(Constraint.RelativeCylinder(radius: 25))
+                .FilterResults(new[]
+                {
+                    Position.ComponentId, Metadata.ComponentId, TransformInternal.ComponentId, Collisions.ComponentId,
+                    SpinnerColor.ComponentId, SpinnerRotation.ComponentId, Score.ComponentId
+                });
+
+            var interest = InterestTemplate.Create()
+                .AddQueries<Position.Component>(clientSelfInterest, clientRangeInterest)
+                .AddQueries<Metadata.Component>(serverSelfInterest, serverRangeInterest);
+
+            template.AddComponent(interest.ToSnapshot(), WorkerUtils.UnityGameLogic);
+
+            TransformSynchronizationHelper.AddTransformSynchronizationComponents(template, clientAttribute);
+            PlayerLifecycleHelper.AddPlayerLifecycleComponents(template, clientWorkerId, WorkerUtils.UnityGameLogic);
+
+            template.SetComponentWriteAccess(EntityAcl.ComponentId, WorkerUtils.UnityGameLogic);
+
+            return template;
+        }
+
+        public static EntityTemplate CreateCubeEntityTemplate(Vector3 location)
```

Comment:
How does this work in multi-worker? If there are no players there's no effective area of interest for the server around a cube. Meaning the cube sudden pops into the worker when it moves across the worker bounday.

Reply:
That's correct, but each cube entity doesn't care about other cubes

And I think transform sync only does the collisions on authoritative cubes anyway

Reply:
> And I think transform sync only does the collisions on authoritative cubes anyway

🤔